src/Domain/BlockNeighbor.hpp

```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template BlockNeighbor.
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+#include <utility>
+
+#include "Domain/Direction.hpp"
+#include "Domain/Orientation.hpp"
+
+/// \ingroup ComputationalDomain
+/// Information about the neighbor of a Block in a particular direction.
+///
+/// \tparam VolumeDim the volume dimension.
+template <size_t VolumeDim>
+class BlockNeighbor {
+ public:
+  BlockNeighbor() = default;
+
+  /// Construct with the Id and orientation of the neighbor.
+  ///
+  /// \param id the Id of the neighbor.
+  /// \param orientation the Orientation of the neighbor.
+  BlockNeighbor(size_t id, Orientation<VolumeDim> orientation);
+  ~BlockNeighbor() = default;
+  BlockNeighbor(const BlockNeighbor<VolumeDim>& neighbor) = default;
+  BlockNeighbor(BlockNeighbor<VolumeDim>&&) noexcept = default;  // NOLINT
+  BlockNeighbor<VolumeDim>& operator=(const BlockNeighbor<VolumeDim>& rhs) =
+      default;
+  BlockNeighbor<VolumeDim>& operator=(  // NOLINT
+      BlockNeighbor<VolumeDim>&&) noexcept = default;
+
+  size_t id() const { return id_; }
+
+  const Orientation<VolumeDim> orientation() const { return orientation_; }
```

Comment:
Should this be returning a const reference? If so, then this should also be `noexcept`